Source/DafnyCore/Resolver/PreType/PreTypeResolve.cs

```diff
@@ -614,6 +614,9 @@ bool IsConversionCompatible(DPreType fromType, DPreType toType) {
       if (fromFamily == PreType.TypeNameInt && toName is PreType.TypeNameChar or PreType.TypeNameReal or PreType.TypeNameORDINAL) {
         return true;
       }
+      if (fromFamily == PreType.TypeNameChar && (toFamily is PreType.TypeNameInt or PreType.TypeNameORDINAL || IsBitvectorName(toFamily))) {
```

Comment:
You might not need this any more given the alternate implementation suggestion, but seems fine to include this anyway.

But if you're converting from a `char` to a bitvector, don't you have to check that the bitvector type is wide enough?

Reply:
I'll still need it because it's for char conversion. The bitvector type width is checked at verification time, here this is a resolution error (and it was working that way in the previous type checker).

Reply:
Thanks for clarifying, just being paranoid.